text/0000-RouteInfo-Metadata.md

```diff
@@ -0,0 +1,210 @@
+- Start Date: 2018-11-02
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# RouteInfo MetaData
+
+## Summary
+
+The RFC introduces the ability to associate application specific metadata with it's corresponding `RouteInfo` object. This also adds a `metadata` field to the `RouteInfo` which will be the return value of `buildRouteInfoMetaData` for it's corresponding `Route`.
+
+```js
+// app/route/profile.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+
+  buildRouteInfoMetaData() {
+    return {
+      trackingKey: 'page_profile',
+      profile: {
+        viewing: this.userId,
+        locale: this.userLocale
+      }
+    }
+  }
+  // ...
+});
+```
+
+```js
+// app/services/analytics.js
+import Service, { inject } from '@ember/service';
+
+export default Service.extend({
+  router: inject('router'),
+  init() {
+    this._super(...arguments);
+    this.router.on('routeDidUpdate', (transition) => {
+      let { to, from } = transition;
+      let { metadata: fromMeta } = from;
+      let { metadata: toMeta } = to;
+      ga.sendEvent('pageView', {
+        from: fromMeta,
+        to: toMeta,
+        timestamp: Date.now(),
+      })
+    })
+  },
+  // ...
+});
+```
+
+## Motivation
+
+While the `RouteInfo` object is suffcient in providing developers metadata about the `Route` it self, it is not suffcient in layering on application specific metadata about the `Route`. This metadata could be anything from just a more domain specific name for a `Route` e.g. `profile_page` vs `profile.index`, all the way to providing contextual data when the `Route` was visited.
+
+## Detailed design
+
+### `buildRouteInfoMetaData`
+
+This optional hook is intended to be used as a way of letting the routing system know about any metadata associated with the route. It is called once any time the `Route` is entered.
+
+#### `Route` Inteface Extension
+
+```ts
+interface Route {
+  // ... existing public API
+  buildRouteInfoMetaData(): unknown
+}
+```
+
+### `RouteInfo.metadata`
+
+The `metadata` optional field on the `RouteInfo` will be populated with the return value of `buildRouteInfoMetaData`. If there is no meta data associated with the `Route`, the `metadata` field will be `null`.
+
+```ts
+interface RouteInfo {
+  // ... existing public API
+  metadata: Maybe<unknown>;
+}
+```
+
+This field will also be added to `RouteInfoWithAttributes` as it is just a super-set of `RouteInfo`.
+
+
+## How we teach this
```

Comment:
This should probably mention updating both the API docs and the guides. I believe that the super rentals example does deal with tracking which is _exactly_ what this is helpful for...